proposals/js-promise-integration/Overview.md

```diff
@@ -2,299 +2,225 @@
 
-The complete import object looks like:
-```
+```js
 var init_state = () => 2.71;
 var importObj = {js: {
     init_state: init_state,
-    compute_delta:suspending_compute_delta}};
+    compute_delta: new WebAssembly.Suspending(compute_delta)}};
 ```
-In addition to preparing the import, we must also handle the export side. As with the import, we will use a helper&mdash;`$update_state_export` which has the additional suspender argument. This function takes the suspender&mdash;as an `externref` value&mdash;stores it in the `$suspender` global and then calls our normal `$update_state` function:
 
-```
-    (func $update_state_export (export "update_state_export") 
-      (param $susp externref)(result f64)
-      (local.get $susp)
-      (global.set $suspender)
-      (return_call $update_state)
-    )
-```
-The process of wrapping exports is a little different to wrapping imports; in part because we prepare imports before instantiating modules and the wrapping of the export is done afterwards:
-```
+In addition to preparing the import, we must also handle the export side. The process of wrapping exports is a little different to wrapping imports; in part because we prepare imports before instantiating modules and we wrap exports afterwards:
+
+```js
 var sampleModule = WebAssembly.instantiate(demoBuffer,importObj);
-var update_state = new WebAssembly.function(
-  {parameters:[], results:['externref']},
-  sampleModule.exports.update_state_export,
-  {promising : "first"})
+var promise_update = WebAssembly.promising(sampleModule.exports.update_state)
 ```
-The resulting modified module allows the synchronous style application to operate using asynchronous APIs.
 
-At runtime, a call to the exported `$update_state_export` function results in a call to the `$compute_delta` import. 
+At runtime, a call to the JavaScript function `promise_update` will get a `Promise`. As part of resolving that `Promise`,  the WebAssembly exported `$update_state` function is called, which results in a call to the `$compute_delta` import. That, in turn, uses `fetch` to access a remote file, and parse the result in order to give the actual floating point value back to the WebAssembly module. Since we use the wrapped `suspending_compute_delta` to implement `$compute_delta`, the import call will be suspended.
 
-That, in turn, uses `fetch` to access a remote file, and parse the result in order to give the actual floating point value back to the WebAssembly module. Since `fetch` returns a `Promise`, the import call will be suspended.
+When the `fetch` completes, the result is parsed -- which will likely also cause a suspension since getting the text from a `Response` also results in a `Promise`. This too will cause the application to be suspended; but when that finally is resumed the text is parsed and the result returned as a float to `$compute_delta`.
```

Comment:
Nit: you can remove the "likely" from this sentence since, as you say, the `.text()` method returns a promise.

Reply:
ack.